opentracing-api/src/main/java/io/opentracing/Tracer.java

```diff
@@ -117,6 +119,17 @@
      */
     <C> SpanContext extract(Format<C> format, C carrier);
 
+    /**
+     * Closes the Tracer, and flushes the in-memory collection to the configured persistance store.
+     *
+     * <p>
+     * The close method should be considered idempotent; closing an already closed Tracer should not raise an error.
+     * Spans that are created or finished after a Tracer has been closed may or may not be flushed.
+     * Calling the close method should be considered a synchronous operation.
```

Comment:
What should happen if tracer uses sync span export (e.g. via HTTP), is able to retry failures, and the tracing backend is down? This method theoretically can block for a long time?

Reply:
Oh, that's a good point. Retries are fine, I think - but that brings us to the timeout point, and whether we should tell the user it can block for long or it should return relatively soon.

Opinions on this?

(I _think_ every Tracer should try to flush for some given time -implementation specific- and then simply give up).